### PR TITLE
Bug #181 (1067 - Gitlab): login bug

### PR DESCRIPTION
**Pull request by joao4all**
### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/RHEAGROUP/COMET-WEB-Community-Edition/pulls) open
- [x] I have verified that I am following the COMET-WEB [code style guidelines](https://raw.githubusercontent.com/RHEAGROUP/COMET-WEB-Community-Edition/master/.github/CONTRIBUTING.md)
- [x] I have provided test coverage for my change (where applicable)

### Description

Fixed [#181](https://gitlab.innovetusa.com/comet/1067/-/issues/181) Gitlab bug

Solved problems with login page: now focused components will not show their error message. BindValueMode updated

### CODE REVIEW

**Review by samatrhea — APPROVED** [review]

**Review by antoineatrhea — CHANGES_REQUESTED** [review]
Small documentation missing. Could be nice to increase code coverage by testing cases on the Component

File: COMET.Web.Common/Components/Login.razor [hunk 1]
```diff
@@ -24,35 +24,71 @@
...
+                           NullText="Enter your password"
+                           BindValueMode="BindValueMode.OnInput"
+                           @onfocus="@(() => this.HandleFieldFocus("Password"))"
+                           @onblur="@(() => this.HandleFieldBlur("Password"))"
+                           Password="true"/>
+            </Template>
+        </DxFormLayoutItem>
+    </DxFormLayout>
+
+    <!-- <ValidationSummary/> -->
```
Comment (antoineatrhea) [inline COMET.Web.Common/Components/Login.razor:65]: Remove this commented code please

File: COMET.Web.Common.Tests/Components/LoginTestFixture.cs [hunk 2]
```diff
@@ -70,6 +73,36 @@ public void Teardown()
...
+            var renderer = this.context.RenderComponent<Login>();
+
+            Assert.That(renderer.Instance.FieldsFocusedStatus, Is.EqualTo(new Dictionary<string, bool>()
+            {
+                { "SourceAddress", false },
+                { "UserName", false },
+                { "Password", false }
+            }));
+
+            Assert.IsFalse(renderer.Instance.FieldsFocusedStatus["UserName"]);
```
Comment (antoineatrhea) [inline COMET.Web.Common.Tests/Components/LoginTestFixture.cs:88]: Please use Assert.That and surround the two Assert inside a Assert.Multiple( () => ... );

File: COMET.Web.Common/Components/Login.razor.cs [hunk 3]
```diff
@@ -64,6 +65,20 @@ public partial class Login
...
 
+        /// <summary>
+        /// The dictionary of focus status from the fields, used by blazor component
+        /// </summary>
+        private Dictionary<string, bool> fieldsFocusedStatus;
+
+        /// <summary>
+        /// The dictionary of focus status from the fields
+        /// </summary>
+        public Dictionary<string, bool> FieldsFocusedStatus
```
Comment (antoineatrhea) [inline COMET.Web.Common/Components/Login.razor.cs:76]: Can you remove the private property and just have the public ? and just put {get; private set;} ?

**Review by antoineatrhea — CHANGES_REQUESTED** [review]

**Review by antoineatrhea — APPROVED** [review]

**Pull request merged**